app/services/trade_execution.py

```
@@ -723,76 +724,527 @@ async def _execute_real_order(
+                # Validate required fields before creating Trade
+                if not exchange_account_id:
+                    self.logger.error("Cannot create Trade record: exchange_account_id is required")
+                    return
+                
+                # Convert IDs to proper UUID format
+                try:
+                    user_uuid = uuid.UUID(user_id) if isinstance(user_id, str) else user_id
+                    account_uuid = uuid.UUID(exchange_account_id) if isinstance(exchange_account_id, str) else exchange_account_id
+                    external_order_uuid = uuid.UUID(execution_result["order_id"]) if execution_result.get("order_id") else None
+                except ValueError as e:
+                    self.logger.error("Invalid UUID format for Trade creation", error=str(e))
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)